database-plugins/src/main/java/co/cask/hydrator/plugin/db/batch/source/DBSource.java

```diff
@@ -294,6 +310,16 @@ private void validate() {
       if (!containsMacro("boundingQuery") && (boundingQuery == null || boundingQuery.isEmpty())) {
         throw new IllegalArgumentException("The boundingQuery must be specified if numSplits is not set to 1.");
       }
+
+    }
+
+    private Schema getSchema() {
+      try {
+        return Schema.parseJson(schema);
+      } catch (IOException e) {
+        throw new IllegalArgumentException(
+          String.format("Unable to parse schema '%s'. Reason: %s", schema, e.getMessage()));
```

Comment:
instead of just putting `e.getMessage()` as part of the new exception message, I think you should pass the entire `e` as the second parameter to the constructor of `IllegalArgumentException`.